src/datadog/grok/filters/keyvalue.rs

```diff
@@ -125,27 +179,40 @@ pub fn apply_filter(value: &Value, filter: &KeyValueFilter) -> Result<Value, Gro
     match value {
         Value::Bytes(bytes) => {
             let mut result = Value::Object(BTreeMap::default());
-            parse(
-                String::from_utf8_lossy(bytes).as_ref(),
-                &filter.key_value_delimiter,
-                &filter
-                    .field_delimiters
-                    .iter()
-                    .map(|s| s.as_str())
-                    .collect::<Vec<&str>>(),
-                &filter.quotes,
-                &filter.value_re,
-            )
-            .unwrap_or_default()
-            .into_iter()
-            .for_each(|(k, v)| {
-                if !(v.is_null()
-                    || matches!(&v, Value::Bytes(b) if b.is_empty())
-                    || k.trim().is_empty())
-                {
-                    result.insert(crate::path!(&k), v);
-                }
-            });
+            let value = String::from_utf8_lossy(bytes);
+            filter
+                .re_pattern
+                .captures_iter(value.as_ref())
+                .for_each(|c| {
+                    let key = parse_key(extract_capture(&c, 1), filter.quotes.as_slice());
+                    if key.contains(' ') {
+                        return;
+                    }
```

Comment:
Personally, I'd do this in the inverse just to avoid the early return, but TIOLI:
```suggestion
                    if !key.contains(' ') {
```

Reply:
I prefer an early exit to be honest than having an extra nested level, however given that we can extract the code below into a function(as you suggest) this make sense.